### DISCUSSION TITLE
Add support for sending IMEI information if available

### DISCUSSION THREAD

**Discussion by washingtonuesc** [strongswan/strongswan#214] [Ideas]
Hi Guys,

Would it be possible to add support for sending IMEI information if available as below 3GPP statement extracted from TS 24.302 clause 7.2.6:

`if the UE:
- receives IKE_AUTH response message from ePDG containing a Notify payload with the DEVICE_IDENTITY Notify Message Type and the Identity Type field of the DEVICE_IDENTITY Notify payload is set to either 'IMEI' or 'IMEISV' and the Identity Value field is empty;
- successfully authenticates the network or requests emergency session; and
- has Mobile Equipment Identity IMEI or IMEISV available,

**the UE shall include the DEVICE_IDENTITY Notify payload in the new IKE_AUTH request message**.`

Thank you,
Washington

**Comment by tobiasbrunner** [comment]
You can write your own plugin that handles such notifies.

**Comment by strongX509** [comment]
I couple of years ago I wrote a simple example on how to define a private notify message and create a plugin to set and read it. The source code can be found in the [demo-plugin](https://github.com/strongswan/strongswan/tree/demo-plugin) branch which consists of the single commit https://github.com/strongswan/strongswan/commit/469a449a1ea05ef81338eabb1c6596128dc4b40c

Andreas

**Reply by tobiasbrunner** [reply]
Note that plugins can also be built out-of-tree (i.e. without having to modify the configure script and the Makefiles). An example is the [strongswan-redirect-plugin](https://github.com/tobiasbrunner/strongswan-redirect-plugin) I created a while ago.

**Reply by washingtonuesc** [reply]
Thank you guys for the tips. 
They will be very helpful.
I really appreciate!
